src/libsystemd/sd-id128/id128-util.c

```diff
@@ -210,3 +210,20 @@ sd_id128_t id128_make_v4_uuid(sd_id128_t id) {
 }
 
 DEFINE_HASH_OPS(id128_hash_ops, sd_id128_t, id128_hash_func, id128_compare_func);
+
+int id128_get_product(sd_id128_t *ret) {
+        sd_id128_t uuid;
+        int r;
+
+        r = id128_read("/sys/class/dmi/id/product_uuid", ID128_UUID, &uuid);
+        if (r == -ENOENT)
+                r = id128_read("/sys/firmware/devicetree/base/vm,uuid", ID128_UUID, &uuid);
```

Comment:
the duplicated code below had the comment `/* on POWER, it's exported here instead */` - imho it's worth keeping, as it's a useful notation for readers

Reply:
(unless it's outdated/wrong)